### ISSUE TITLE
"How much did I save?" query

### ISSUE THREAD

**Issue by alexjj** [aumayr/beancount-sql-queries#10]
Thinking about the best way to answer this question in a query...

Thinking maybe select specific accounts and then show their balance on certain dates, but not sure how to get it to do it all together, any ideas?

**Comment by aumayr** [comment]
@alexjj Can you provide a bit more description/example what you want to achieve?

**Comment by adamgibbins** [comment]
I 👍 'd this as I've a similar want. I basically want to measure the percentage of my unspent income over time. i.e. the "savings rate" - so I can ensure I'm always saving x% of my income and have visibility of any regressions on that and can compensate the following month.

The most realistic is probably a bean query with year/month and 'savings rate' as columns.

**Comment by aumayr** [comment]
Wouldn't that just be a monthly aggregated subtraction of `Expense`\- and `Income`-accounts as a percentage of the `Income` (as in the Income Statement)?

**Comment by alexjj** [comment]
> Wouldn't that just be a monthly aggregated subtraction of Expense- and Income-accounts as a percentage of the Income (as in the Income Statement)?

That would work - perhaps adding in payments to other liabilities too. I can't figure out to do that in a single query. 

e.g. A table like:

| Month | Income | Expenses & Liabilities | Total Saved | % Saved |
| --- | --- | --- | --- | --- |
| Jan | 1000 | 700 | 300 | 30 |

**Comment by aumayr** [comment]
@blais Is this doable in bean-query?

**Comment by blais** [comment]
You would need multiple queries, and for the shell to somehow deal with the concept of division of two Inventory objects.

**Comment by aumayr** [comment]
@alexjj [fava](https://github.com/aumayr/fava) does provide a chart for this, but currently no way to download/use this information in any other way.

**Comment by clsacramento** [comment]
Which fava report can do that?
I am also interested to seeing this in fava or with query but NetProfit only gives the absolute amount of money I save i.e 50K not a percentage...

I created this query as a start:
SELECT
  year, sum(number) as Total, ROOT(account,1) as type, currency
FROM
    year >= 2018
WHERE
    account ~ 'Expenses|Income'
GROUP BY year, type, currency
ORDER BY year, type

But it gives me a table like this:

year | total | type | currency
-- | -- | -- | --
2018 | 1000 | Expenses | USD
2018 | -2000 | Income | USD
2019 | 1000 | Expenses | USD
2019 | -3000 | Income | USD
2020 | 500 | Expenses | USD
2020 | -4000 | Income | USD

I did not find a way to perform math operations on the two accounts number. Is there anyway to achieve this with either fava or the query language? I would also be interested to see breakdowns for this per year and per month...
